libs/json-api-nestjs/src/lib/mixin/service/transform/transform.mixin.ts

```diff
@@ -186,10 +186,17 @@ export class TransformMixinService<T> {
         const propsData = data[field];
         const typeName = getEntityName(this.relationTarget.get(field));
         if (Array.isArray(propsData)) {
-          builtData.data = propsData.map((i) => ({
-            type: camelToKebab(typeName),
-            id: i[this.relationPrimaryField.get(field)].toString(),
-          }));
+          builtData.data = propsData
```

Comment:
I think will be better like this:

```
const tmp = [].reduce((acum, item) => {
  const key = `${camelToKebab(typeName)}:${i[
    this.relationPrimaryField.get(field)
  ].toString()}`;

  if (acum.has(key)) {
    acum.set(key, {
      type: camelToKebab(typeName),
      id: i[this.relationPrimaryField.get(field)].toString(),
    });
  }

  return acum;
}, new Map());

builtData.data = [...tmp.values()];

```

because we will use only one iterate otherwise we will have 2 iterate(map and reduce) and inner iterate(find)  for every call reduce callback

Reply:
Agreed. Fixed as per instructions.